File: scripts/notion_to_content/tables.py
from __future__ import annotations


def escape_cell(s: str) -> str:
    s = s.replace("\n", " ").replace("\r", " ")
    s = s.replace("|", "\\|")
    return s


def gfm_table(headers: list[str], rows: list[list[str]]) -> str:
    w = len(headers)
    if w == 0:
        return ""
    lines = [
        "| " + " | ".join(escape_cell(c) for c in headers) + " |",
        "| " + " | ".join("---" for _ in headers) + " |",
    ]
    for r in rows:
        rr = [r[i] if i < len(r) else "" for i in range(w)]
        lines.append("| " + " | ".join(escape_cell(c) for c in rr) + " |")
    return "\n".join(lines) + "\n"

File: packages/_archive/notion-importer/src/tables.ts
export function escapeCell(s: string): string {
  return s.replace(/\n/g, " ").replace(/\r/g, " ").replace(/\|/g, "\\|");
}

export function gfmTable(headers: string[], rows: string[][]): string {
  const w = headers.length;
  if (w === 0) return "";
  const lines = [
    `| ${headers.map((c) => escapeCell(c)).join(" | ")} |`,
    `| ${headers.map(() => "---").join(" | ")} |`,
  ];
  for (const r of rows) {
    const rr = Array.from({ length: w }, (_, i) => (i < r.length ? r[i]! : ""));
    lines.push(`| ${rr.map((c) => escapeCell(c)).join(" | ")} |`);
  }
  return `${lines.join("\n")}\n`;
}
